src/allmydata/storage/http_server.py

```diff
@@ -539,7 +539,7 @@ def read_share_chunk(self, request, authorization, storage_index, share_number):
     )
     def add_or_renew_lease(self, request, authorization, storage_index):
         """Update the lease for an immutable share."""
-        if not self._storage_server.get_buckets(storage_index):
+        if not list(self._storage_server._get_bucket_shares(storage_index)):
```

Comment:
This looks like it's an optimization?  Or I'm missing what the change in behavior is.

It'd be nice to have a public API, even if just an internal one, and stick to it.  So if this can reasonably still be `get_buckets` then can you switch it back to that?  Or if not ... let's make `_get_bucket_shares` public?  Or introduce a new public method explicitly for this question - something like `_storage_server.bucket_exists(storage_index)`?

Reply:
No, it's to make it work for both mutables and immutables. The API naming in server.py isn't great...